arelle/plugin/validate/HMRC/ValidateHmrc.py

```diff
@@ -712,12 +712,13 @@ def validate(self) -> None:
                     self.validateUnauditedLLPGroupAccounts()
                 else:
                     self.validateUnauditedCompanyGroupAccounts()
-            elif (self.applicableLegislation == CONCEPT_SMALL_COMPANY_REGIME_FOR_ACCOUNTS
-                  and not (self.scopeAccounts in {
-                        ScopeAccounts.GROUP_ONLY.value,
-                        ScopeAccounts.CONSOLIDATED_GROUP.value,
-                    } and self.accountsType == CONCEPT_ABRIDGED_ACCOUNTS)):
-                if self.legalFormEntity == CONCEPT_LLP:
+            elif not (
+                self.scopeAccounts in {
+                    ScopeAccounts.GROUP_ONLY.value,
+                    ScopeAccounts.CONSOLIDATED_GROUP.value,
+                } and self.accountsType == CONCEPT_ABRIDGED_ACCOUNTS
```

Comment:
It seems like this `not ...` is always true, so this whole `elif` is equivalent to `else`?  If `scopeAccounts in {GROUP_ONLY,CONSOLIDATED_GROUP}`, that scenario is already captured by the `elif` on line 707.  The original logic also looks wrong in that way.  Am I misreading the logic?

Do you have any more detail on what issue we're trying to fix or the authority for this logic?

Reply:
The main issue is that Applicable Legislation should not be checked for small companies or LLPs unless the ScopeAccount and AccountsType meet the requirements. In that case we want to check Applicable Legislation and divert to LLP if it passes otherwise do nothing.